Firestore/Example/Tests/Integration/API/FIRServerTimestampTests.m

```diff
@@ -88,14 +103,46 @@ - (void)writeInitialData {
   XCTAssertEqualObjects(initialDataSnap.data, _initialData);
 }
 
+/** Waits for a snapshot with local writes. */
+- (FIRDocumentSnapshot *)waitForLocalEvent {
+  return [_accumulator awaitEventWithName:@"Local event."];
+}
+
 /** Waits for a snapshot containing _setData but with NSNull for the timestamps. */
-- (void)waitForLocalEvent {
-  FIRDocumentSnapshot *localSnap = [_accumulator awaitEventWithName:@"Local event."];
+- (FIRDocumentSnapshot *)waitForLocalEventWithNullTimestamps {
+  FIRDocumentSnapshot *localSnap = [self waitForLocalEvent];
   XCTAssertEqualObjects(localSnap.data, [self expectedDataWithTimestamp:[NSNull null]]);
+  return localSnap;
+}
+
+/** Waits for a snapshot containing _setData but with a local estimate for the timestamps. */
+- (FIRDocumentSnapshot *)waitForLocalEventWithEstimatedTimestamps {
+  FIRDocumentSnapshot *localSnap = [self waitForLocalEvent];
+  id timestamp = [localSnap valueForField:@"when" options:_estimateValue];
+  XCTAssertTrue([timestamp isKindOfClass:[NSDate class]]);
+  XCTAssertEqualObjects([localSnap dataWithOptions:_estimateValue],
+                        [self expectedDataWithTimestamp:timestamp]);
+  return localSnap;
+}
+
+/** Waits for a snapshot containing _setData but using the previous field value for the timestamps. */
+- (FIRDocumentSnapshot *)waitForLocalEventWithPreviousDataFromSnapshot:
+    (FIRDocumentSnapshot *_Nullable)previousSnapshot {
```

Comment:
See comment above, but if you keep this method (or one like it), it should be `(nullable FIRDocumentSnapshot *)`

Reply:
Fixed.